app/controllers/v0/messages_controller.rb

```diff
@@ -67,8 +68,55 @@ def thread
 
     private
 
+    def get_folder(folder_id)
+      folders = client.get_folder(folder_id)
+
+      raise VA::API::Common::Exceptions::RecordNotFound, folder_id unless folders.present?
+      folders.data.first
+    end
+
+    # If page is not supplied, we are bringing back all results in the set. In that case
+    # adjust the per_page count to the TBD maximum number of messages allowed per MHV call
+    def pagination_params
+      folder_id = params[:folder_id].try(:to_i)
+      folder = get_folder(folder_id)
+      page = params[:page].try(:to_i)
+
+      if page.present?
+        pages = page..page
+        per_page = [params[:per_page].try(:to_i) || DEFAULT_PER_PAGE, MAXIMUM_PER_PAGE].min
+      else
+        # We need to test concatenation of multiple message GETS.
+        # Use only maximum per page once we can create more messages in a folder.
+        per_page = params[:per_page].try(:to_i) ||  MAXIMUM_PER_PAGE
+        pages = 1..(folder.count.to_f / per_page).ceil
+      end
+
+      { folder_id: folder_id, pages: pages, per_page: per_page, count: folder.count }
+    end
+
     def message_params
-      params.permit(:id, :category, :body, :recipient_id, :subject, :format)
+      # ActionController::Parameters No Longer Inherits from HashWithIndifferentAccess
+      # Gem message api uses keyword arguments and will not work with HashWithIndifferentAccess according
+      # to longstanding bug. Allegedly was fixed in Ruby 2.2, but having same issue in Ruby 2.3
+      hash = params.permit(:id, :category, :body, :recipient_id, :subject).to_h
+      Hash[hash.map { |k, v| [k.to_sym, v] }]
+    end
+
+    # Unwraps data from individual calls to MHV and aggregates the results to a new collection,
+    # potentially containing all messages in the folder
+    def get_messages(folder_id, pagination)
+      data = []
+      pagination[:pages].each do |page|
+        data.concat(client.get_folder_messages(folder_id, page, pagination[:per_page]).attributes)
+      end
+
+      meta = {
```

Comment:
It can be, and I will do that, but wanted to get this in now. I will handle both the issue with accepting symbol/string keywords and concatenating results sets then.